### ISSUE TITLE
REST API Usage - Empty reply following documentation

### ISSUE THREAD

**Issue by Yoshyn** [freqtrade/freqtrade#3820]
## Describe your environment

  * Operating system: Using Mac with Docker
  * Freqtrade Version: freqtrade docker-d2111c08
  
## Your question

TLDR : I'm not able to make the ping to the API following the documenation. Got an Empty reply from server.

I'm using the [Docker without docker-compose
](https://www.freqtrade.io/en/latest/docker) and [the minimal configuration for Docker](https://www.freqtrade.io/en/latest/rest-api/) related to API.

My config is a sample one create with new-config. I just update the api_server part :
```
  "max_open_trades": 3,
  "stake_currency": "BTC",
  "stake_amount": 0.07,
  "tradable_balance_ratio": 0.99,
  "fiat_display_currency": "USD",
  "timeframe": "1m",
  "dry_run": true,
  ...
  "api_server": {
      "enabled": true,
      "listen_ip_address": "0.0.0.0",
      "listen_port": 8080
  },
```

I run `freqtrade` almost the same than in the documentation (only mounting point, and option to freqtrade change)
```
docker run --name freqtrade \
  -v "~/.freqtrade/user_data:/freqtrade/user_data" \
  -p 127.0.0.1:8080:8080 \
  freqtrade trade --db-url sqlite:////freqtrade/user_data/tradesv3.sqlite --config /freqtrade/user_data/config.json --strategy MaSuperStrategy
```

Then, no error, the process run well and stay alive without error.
At the beginning, I can even grep the apiserver into log : 
```
2020-10-02 14:28:03,683 - freqtrade.rpc.api_server - INFO - Starting HTTP Server at 127.0.0.1:8080
....
2020-10-02 14:28:03,685 - freqtrade.rpc.api_server - INFO - Starting Local Rest Server.
```

The port mapping seem to work : 
```
❯❯❯ docker ps -a
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                      NAMES
5a95f9d9dc60        freqtrade           "freqtrade trade --d…"   55 seconds ago      Up 54 seconds       127.0.0.1:8080->8080/tcp   freqtrade
```

So at this point, I expect to be able to call `/api/v1/ping` and get a `{"status":"pong"}`.

But instead, on my host, if I try to make a curl :
```
curl http://127.0.0.1:8080/api/v1/ping
curl: (52) Empty reply from server
```

Is there something I miss in the configuration ?

**Comment by xmatthias** [comment]
I don't think you're using the right config file... (not the one you're showing above, anyway).

You've set the `listen_ip_address` to `0.0.0.0` (which is correct for docker).

However, the API is starting at 127.0.0.1 (see below) - which is localhost within the docker container - so even though the port mapping into the container works - there's nothing listening on the inside.

```
2020-10-02 14:28:03,683 - freqtrade.rpc.api_server - INFO - Starting HTTP Server at 127.0.0.1:8080
```

You might be able to execute bash within the container from another shell (`docker exec --entrypoint bash <containerid/name>`)  and run the curl-command there (it should work from within the container - showing that the API is indeed starting) - but the log must say "Starting HTTP Server at 0.0.0.0:8080" ...

**Comment by Yoshyn** [comment]
Ok, afterwards my question was stupid but I don't regret to have asked it because I was looking completely aside...
Thanks a thousand times. It was exactly that. A bad configuration file
